### PR TITLE
Feature/aupro test

### LINKED ISSUE #443: Add unit-tests for `AUPRO` metric
**Is your feature request related to a problem? Please describe.**
  - In order to maintain good code health, its important to add tests to it. Currently, the AUPRO metric is not being tested.

**Describe the solution you'd like**
  - Add unit-tests for AUPRO metric

**Additional context**
  - AUPRO was added in #429

### PR DESCRIPTION
**Pull request by ORippler**
# Description

- Provide a summary of the modification as well as the issue that has been resolved. List any dependencies that this modification necessitates.
  -  Adds unit-tests for `AUPRO` metric. While writing the tests, edge-cases were discovered and smaller bugs fixed.

- Fixes #443 

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

--------
Remarks:

The reference implementation is taken from [here](https://github.com/eliahuhorwitz/3D-ADS/blob/main/utils/au_pro_util.py), and currently unlicensed. The code itself is an adaptation of the official reference-code provided by MVTec.

If this prevents merging the PR, I would suggest removing the code and the corresponding tests from the unit-tests.

### CODE REVIEW

**Review by samet-akcay — APPROVED** [review]
Thanks @ORippler for adding this! 

The test scripts do not seem to follow the code-quality standards, but it's ok for now. We don't check the tests at the moment, which we need to refactor quite soon.

File: tests/helpers/aupro_reference.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
+# REMARK: CODE WAS TAKEN FROM https://github.com/eliahuhorwitz/3D-ADS/blob/main/utils/au_pro_util.py
```
Comment (ashwinvaidya17) [inline tests/helpers/aupro_reference.py:1]: We should also add a license file next to this similar to https://github.com/openvinotoolkit/anomalib/blob/development/anomalib/models/draem/LICENSE. This should include a copy of the license contained in https://www.mydrive.ch/shares/45924/9ce7a138c69bbd4c8d648b72151f839d/download/428846918-1643297332/evaluation_code.tar.xz. However, I am more concerned about this repo https://github.com/eliahuhorwitz/3D-ADS as it does not have a license. Maybe it would be a good idea to request the authors to add a license to their repo.

**Review by ashwinvaidya17 — CHANGES_REQUESTED** [review]
Thanks for making this PR! I have a single concern regarding the license.

File: tests/helpers/aupro_reference.py (hunk 1 above)
Comment (ORippler) [reply tests/helpers/aupro_reference.py:1]: I added the required licenses, but please do double check.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks!

**Review by samet-akcay — APPROVED** [review]
Thanks a lot!
